### ISSUE TITLE
About loss, image_loss when train model?

### ISSUE THREAD

**Issue by vvthai10** [zqhang/AnomalyCLIP#16]
I train with dataset visa,
I train 11 epoch, but loss and image_loss is 3.7960, 0.5325. I feel something wrong. I setup by your setting. Can you share me about loss and image_loss when u train?

**Comment by wyy123de** [comment]
![image](https://github.com/zqhang/AnomalyCLIP/assets/145307001/904f651f-e3d3-4eba-8889-8450c71a0913)
我的损失比图像损失还小

**Comment by vvthai10** [comment]
Dataset Brain, you train by this?? What dataset u use to train model? And
can u share about hyper param batch, epoch, learning_rate?

**Comment by vvthai10** [comment]
and how can you show the table after train 15 epochs, I not find way to show this

**Comment by wyy123de** [comment]
![image](https://github.com/zqhang/AnomalyCLIP/assets/145307001/72ea8e9b-104d-4eac-94cc-138ea7b80fd6)

**Comment by wyy123de** [comment]
这个你要自己保存到results中，我用的Head-CT数据集
还可以可视化呢

**Comment by wyy123de** [comment]
![image](https://github.com/zqhang/AnomalyCLIP/assets/145307001/e218b5af-967d-4ef9-8a5a-ce5a1b86adfd)

**Comment by vvthai10** [comment]
You train by Brain dataset, not mvtec

**Comment by vvthai10** [comment]
Can u share me code you visualize graphwhen train model

**Comment by wyy123de** [comment]
对啊，我是医学影响异常检测

**Comment by wyy123de** [comment]
sure

**Comment by vvthai10** [comment]
I'm also studying the medical part like you, learnign_rate u still 0.001 ?

**Comment by wyy123de** [comment]
yes

**Comment by vvthai10** [comment]
can u share me link of dataset u use,

**Comment by vvthai10** [comment]
I try train with mvtec, after 15 epoch, loss, and image_loss is 3.4, 0.3 but when test result still okay

**Comment by wyy123de** [comment]
嗯嗯[HeadCT](https://www.kaggle.com/datasets/felipekitamura/head-ct-hemorrhage)

**Comment by wyy123de** [comment]
我感觉我训练有BUG

**Comment by wyy123de** [comment]
[HeadCT](https://www.kaggle.com/datasets/felipekitamura/head-ct-hemorrhage)

**Comment by wyy123de** [comment]
parser.add_argument("--train_data_path", type=str, default="/root/Downloads/Untitled Folder/AnomalyCLIP-main", help="train dataset path")
    parser.add_argument("--save_path", type=str, default='/root/Downloads/Untitled Folder/AnomalyCLIP-main/results', help='path to save results')
这个是保存代码，其实作者给的代码很详细，你要保存到位置就可以看到

**Comment by vvthai10** [comment]
And now i want to add adapter to image encoder, u want try with me

**Comment by wyy123de** [comment]
![image](https://github.com/zqhang/AnomalyCLIP/assets/145307001/750ece66-fb9e-4a6d-b780-e9c65d92a4cc)
训练出现了BUG

**Comment by wyy123de** [comment]
导致效果好

**Comment by wyy123de** [comment]
![image](https://github.com/zqhang/AnomalyCLIP/assets/145307001/676d432e-d781-4b6f-9568-425cb67d886b)

**Comment by vvthai10** [comment]
你的意思是你的结果比文章高很多吧？

**Comment by wyy123de** [comment]
对

**Comment by vvthai10** [comment]
看起来他们在文章中使用 mvtec 进行训练并在医疗设备上进行评估，但如果你使用医疗包进行训练，我认为结果会有所不同。

**Comment by wyy123de** [comment]
![image](https://github.com/zqhang/AnomalyCLIP/assets/145307001/40f12110-713b-4341-8e14-04d7b3677717)
跑了100个epoch

**Comment by vvthai10** [comment]
为什么训练了 100 个 epoch，结果却低于 15 个 epoch？

**Comment by vvthai10** [comment]
在数据集文件中，我没有看到的generate_class_info函数返回大脑数据集的obj_list

**Comment by wyy123de** [comment]
100个epoch，训练的多，我学习率调到0.000001，15个epoch学习率0.001

**Comment by vvthai10** [comment]
我发现在这种类型的练习中，他们仅训练 15 或 50 个 epoch，lr 为 0.001 或 0.0001。 因为训练数据很小，训练多了就怕过拟合

**Comment by wyy123de** [comment]
yes

**Comment by vvthai10** [comment]
但是你的损失确实很低，我不明白为什么，因为当我用mvtec文件训练时，它是3.x，哈哈哈

**Comment by vvthai10** [comment]
您想与您的团队就这个主题进行合作吗？ 我正在尝试更新图像编码器部分以将其合并到本文中。

**Comment by wyy123de** [comment]
可能你才是对的，不过损失不下降，可能是欠拟合，你可以尝试调大epoch，或者学习率

> 但是你的损失确实很低，我不明白为什么，因为当我用mvtec文件训练时，它是3.x，哈哈哈

**Comment by wyy123de** [comment]
好啊，我很荣幸



---Original---
From: ***@***.***&gt;
Date: Tue, May 28, 2024 16:54 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [zqhang/AnomalyCLIP] About loss, image_loss when train model?(Issue #16)




 
您想与您的团队就这个主题进行合作吗？ 我正在尝试更新图像编码器部分以将其合并到本文中。

**Comment by vvthai10** [comment]
您可以通过电子邮件与我联系 ***@***.***

**Comment by wyy123de** [comment]
ok

**Comment by vvthai10** [comment]
请给我发电子邮件，我会详细告诉你我在做什么:)

**Comment by wyy123de** [comment]
![8f0b24d7f78238e87fa972c1d079697](https://github.com/zqhang/AnomalyCLIP/assets/145307001/0862f8d7-71a1-44aa-97f8-05803c242479)
是这个邮箱吗

**Comment by vvthai10** [comment]
yeah, i ssee
